### PR TITLE
EN-8181/ReturnData-to-transfer-for-callback

### PR DESCRIPTION
**Pull request by sasurobert**
Added new flags and fixed a scenario which was not treated the same way as in arwen.

### CODE REVIEW

File: cmd/node/config/config.toml [hunk 1]
```diff
@@ -52,6 +52,12 @@
    # BlockGasAndFeesReCheckEnableEpoch represents the epoch when gas and fees used in each created or processed block are re-checked
    BlockGasAndFeesReCheckEnableEpoch = 4
 
+   # ReturnDataToLastTransferEnableEpoch represents the epoch when return data is added to last output transfer for callbacks
```
Comment (SebastianMarian) [inline cmd/node/config/config.toml:55]: when returned

File: process/smartContract/process.go [hunk 2]
```diff
@@ -1808,7 +1812,7 @@ func (sc *scProcessor) useLastTransferAsAsyncCallBackWhenNeeded(
 	tx data.TransactionHandler,
 	result *smartContractResult.SmartContractResult,
 ) bool {
-	if len(vmOutput.ReturnData) > 0 {
+	if len(vmOutput.ReturnData) > 0 && sc.flagReturnDataToLastTransfer.IsSet() {
```
Comment (SebastianMarian) [inline process/smartContract/process.go:1815]: !sc.flagReturnDataToLastTransfer.IsSet() ?

File: cmd/node/config/config.toml [hunk 3]
```diff
@@ -52,6 +52,12 @@
...
+   ReturnDataToLastTransferEnableEpoch = 5
+
+   # ArwenESDTFunctionsEnableEpoch represents the epoch the new ESDT functions are enabled in arwen
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply cmd/node/config/config.toml:58]: done

File: cmd/node/config/config.toml (hunk 1 above)
Comment (sasurobert) [reply cmd/node/config/config.toml:55]: done

File: process/smartContract/process.go (hunk 2 above)
Comment (sasurobert) [reply process/smartContract/process.go]: done

**Review by SebastianMarian — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**